src/App.tsx

```diff
@@ -1,51 +1,117 @@
-import React from 'react';
+import React, { Component } from 'react';
 import './App.scss';
 
-// import goodsFromServer from './goods';
-
-export const App: React.FC = () => {
-  return (
-    <main className="App">
-      <header className="App__header">
-        <h1 className="App__title">
-          Jam is selected
-        </h1>
-
-        <button
-          type="button"
-          className="App__clear"
-        >
-          Clear
-        </button>
-      </header>
-
-      <ul>
-        <li className="Good">Dumplings</li>
-        <li className="Good">Carrot</li>
-        <li className="Good">Eggs</li>
-        <li className="Good">Ice cream</li>
-        <li className="Good">Apple</li>
-        <li className="Good">Bread</li>
-        <li className="Good">Fish</li>
-        <li className="Good">Honey</li>
-        <li className="Good Good--active">Jam</li>
-        <li className="Good">Garlic</li>
-      </ul>
-
-      {/* Put required buttons into each Good */}
-      <button
-        type="button"
-        className="Good__remove"
-      >
-        Remove
-      </button>
-
-      <button
-        type="button"
-        className="Good__select"
-      >
-        Select
-      </button>
-    </main>
-  );
+import goodsFromServer from './goods';
+
+type Product = string;
+
+type State = {
+  selectedGood: Product,
+  selectedGoods: Product[],
 };
+
+export class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    selectedGood: 'Jam',
+    selectedGoods: ['Jam'],
+  };
+
+  onMouseClickRemove = (event: React.MouseEvent<HTMLButtonElement>) => {
+    const clickedButton = event.currentTarget;
+    const parentLi = clickedButton.closest('li');
+
+    parentLi?.classList.remove('Good--active');
+    const text = parentLi?.childNodes[0].nodeValue;
+
+    if (text) {
+      this.state.selectedGoods.forEach((item, i) => {
+        if (item === text) {
+          this.state.selectedGoods.splice(i, 1);
+        }
+      });
+    }
+
+    this.setState({
+      selectedGood: '',
+    });
+  };
+
+  render() {
+    const { selectedGood, selectedGoods } = this.state;
+    let goodsToShow = '';
+    const slicedWord = selectedGoods.slice(0, -1).join(', ');
+    const word = `${slicedWord} and ${selectedGoods[selectedGoods.length - 1]}`;
+
+    if (selectedGoods.length === 1) {
+      goodsToShow += selectedGoods[0];
+    } else if (selectedGoods.length === 2) {
+      goodsToShow += `${selectedGoods[0]} and ${selectedGoods[1]}`;
+    } else {
+      goodsToShow += word;
+    }
```

Comment:
create some method, i don't understand what is it

Reply:
fixed